### PR TITLE
Test related posts

### PR DESCRIPTION
**Pull request by grapesmoker**
This PR adds a test for the `related_posts` function of the CFGOVPage class. In addition, I have substantially refactored the function itself, removing some redundant code and cleaning it up. 

## Changes

- Refactored `related_posts`

## Testing

- Tests for `related_posts`

## Review

@richaagarwal @willbarton @chosak 

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: cfgov/v1/models/base.py [hunk 1]
```diff
@@ -158,46 +158,49 @@ def generate_view_more_url(self, request):
...
+
+        def fetch_children_by_specific_category(block, parent_slug):
+            # This used to be a Page.objects.get, which would throw
+            # an exception if the requested parent wasn't found. As of
+            # Django 1.6, you can now do Page.objects.filter().first();
+            # the advantage here is that you can check for None right
+            # away and not have to rely on catching exceptions, which
+            # in any case didn't do anything useful other than to print
+            # an error message. Instead, we just return an empty query
+            # which has no effect on the final result.
```
Comment (richaagarwal) [inline cfgov/v1/models/base.py]: Minor thing but this comment could be a multiline docstring e.g. """ comment """

File: cfgov/v1/tests/models/test_related_posts.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        # you can just delete the parent and it cascades
+
+        self.blog_parent.delete()
+        self.newsroom_parent.delete()
+        self.events_parent.delete()
+        self.archive_events_parent.delete()
+
+        self.page_with_authors.delete()
+
+    def test_related_posts_blog(self):
```
Comment (richaagarwal) [inline cfgov/v1/tests/models/test_related_posts.py]: Would it be possible to make this test name more descriptive, or describe it in a docstring? ditto for the rest of the tests. There are multiple assertions here and I'm not sure which each one is testing. e.g. what is `self.assertEqual(len(related_posts['Blog']), 2)` testing?  why do we expect it to be 2?

Comment (grapesmoker) [inline cfgov/v1/tests/models/test_related_posts.py]: I added docstrings describing all the tests.

File: cfgov/v1/models/base.py (hunk 1 above)
Comment (grapesmoker) [inline cfgov/v1/models/base.py]: Noted and changed.

**Comment by richaagarwal** [comment]
Thanks for adding the documentation, @grapesmoker ! I think at some point we'll want to break these up into even smaller unit tests, but I realize the code itself may make it hard to test so that may require some refactoring. I'm going to go ahead and merge this.

**Pull request merged**